File: src/main/java/net/kodehawa/lib/imageboards/entities/impl/GelbooruImage.java
package net.kodehawa.lib.imageboards.entities.impl;

import com.fasterxml.jackson.annotation.JsonProperty;
import net.kodehawa.lib.imageboards.entities.BoardImage;
import net.kodehawa.lib.imageboards.entities.Rating;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class GelbooruImage implements BoardImage {
    private String source;
    private String hash;
    private int height;
    private int width;
    private int id;
    private String image;
    private String owner;
    private Rating rating;
    private String tags;
    private String file_url; //Thanks gelbooru for giving a full url I love you
    private int score;
    @JsonProperty("has_children")
    private boolean has_children;
    @JsonProperty("change")
    private long change; // timestamp in seconds

    public String getSource() {
        return source;
    }

    public String getHash() {
        return hash;
    }

    public int getId() {
        return id;
    }

    public String getImage() {
        return image;
    }

    public String getOwner() {
        return owner;
    }

    public String getFile_url() {
        return file_url;
    }

    public boolean isHas_children() {
        return has_children;
    }

    @Override
    public Rating getRating() {
        return rating;
    }

    @Override
    public List<String> getTags() {
        return Collections.unmodifiableList(Arrays.asList(tags.split(" ")));
    }

    @Override
    public int getScore() {
        return score;
    }

    @Override
    public int getHeight() {
        return height;
    }

    @Override
    public int getWidth() {
        return width;
    }

    @Override
    public String getURL() {
        return getFile_url();
    }

    @Override
    public boolean hasChildren() {
        return isHas_children();
    }

    // Doesn't implement this
    @Override
    public boolean isPending() {
        return false;
    }

    @Override
    public long getCreationMillis() {
        return change * 1000;
    }
}

File: src/main/kotlin/net/kodehawa/lib/imageboards/entities/impl/GelbooruImage.kt
package net.kodehawa.lib.imageboards.entities.impl

import net.kodehawa.lib.imageboards.entities.BoardImage
import net.kodehawa.lib.imageboards.entities.Rating

class GelbooruImage : BoardImage {
	val source: String? = null
	val hash: String? = null
	override val height = 0
	override val width = 0
	val id = 0
	val image: String? = null
	val owner: String? = null
	override val rating: Rating? = null
	override val tags: List<String>
		get() = arrayListOf()

	val fileUrl //Thanks gelbooru for giving a full url I love you
			: String? = null

	override val score = 0


	override val uRL: String?
		get() = fileUrl
}